src/services/__tests__/profileService.test.ts

```diff
@@ -193,4 +214,18 @@ describe("ProfileService#upsertProfile", () => {
       expect(e).toEqual(new Error("Api error."));
     }
   });
+
+  it("fails to create-update a user profile returns an 429 HTTP error from the upstream API", async () => {
+    mockGetProfile.mockImplementation(() =>
+      t.success(tooManyReqApiMessagesResponse)
+    );
+
+    const service = new ProfileService(api);
+
+    try {
+      await service.upsertProfile(mockedUser, upsertRequest);
+    } catch (e) {
```

Comment:
this test passes simply because this `catch` block gets never executed

Reply:
Fix in fefbe24